safekeeper/src/bin/safekeeper.rs

```diff
@@ -434,6 +435,12 @@ async fn start_safekeeper(conf: SafeKeeperConf) -> Result<()> {
     let mut sigint_stream = signal(SignalKind::interrupt())?;
     let mut sigterm_stream = signal(SignalKind::terminate())?;
 
+    // Notify systemd that we are ready. This is important as currently loading
+    // timelines takes significant time (~30s in busy regions).
+    if let Err(e) = sd_notify::notify(true, &[NotifyState::Ready]) {
+        warn!("systemd notify failed: {:?}", e);
+    }
+
```

Comment:
How is the init ready at this point...? It is at least unclear to me where is the actual loading and startup, and why is this not at the end of that.

Reply:
Oki I had misunderstood what was needed; this is for rolling sequencial restarts, not a problem of incoherent responses while loading.

Loading is done in the `GlobalTimelines::init`.